lib/video/cubit/video_cubit.dart

```diff
@@ -55,15 +55,34 @@ class VideoCubit extends Cubit<VideoState> {
       threshold: 150,
     );
 
+    final nearestTurnActionTimeStamp = turnActionTimeStampsList?.nearestIndex(
+      target: position.inMilliseconds,
+      threshold: 600,
+    );
```

Comment:
to make sure we show all of these and computational time is less, handling this as a fade-out widget would be best. see example below:
```dart
class _TurnActionDetectionState extends State<TurnActionDetection>
    with TickerProviderStateMixin {
  late final AnimationController _controller = AnimationController(
    duration: const Duration(milliseconds: 800),
    vsync: this,
  );
  late final Animation<double> _animation = CurvedAnimation(
    parent: _controller,
    curve: Curves.easeOut,
  );

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return FadeTransition(
        opacity: _animation,
        child: // our code here
    );
  }
}
```